<h2 style="text-align: center; background-color: #4d648d; font-family:Arial; color: white; padding: 13px; line-height: 1; border-radius:10px">Initiation Python - Import de fichiers 💻</h2>

# Exercice d'introduction

Faire de la veille sur ces sujets et synthétiser vos recherches dans un support de présentation.

Typologie de données :
 - Structurées vs non structurées : définition, exemples, ...
 - Les différents sources de données. Trouvez les packages en Python qui permet notamment de pouvoir importer/se connecter aux données.  

In [ ]:
# Importer les bibliothèques nécessaires
from google.colab import drive
import pandas as pd

# Monter Google Drive
drive.mount('/content/drive')

# Chemin du fichier dans Google Drive (exemple)
csv_file_path = '/content/drive/MyDrive/nom_du_fichier.csv'  # Remplacez 'nom_du_fichier.csv' par le nom de votre fichier
txt_file_path = '/content/drive/MyDrive/nom_du_fichier.txt'  # Remplacez 'nom_du_fichier.txt' par le nom de votre fichier

# Lire un fichier CSV
df_csv = pd.read_csv(csv_file_path)
print("Contenu du fichier CSV :")
print(df_csv.head())

# Lire un fichier TXT
# Spécifiez le délimiteur si nécessaire, ici on utilise un espace comme exemple
df_txt = pd.read_csv(txt_file_path, delimiter=' ')
print("\nContenu du fichier TXT :")
print(df_txt.head())


<h2 style="text-align: center; background-color: #87c7ec; font-family:Arial; color: white; padding: 13px; line-height: 1; border-radius:10px">MISE EN PRATIQUE</h2>

### Import fichier


Import des fichiers et stockage de manière propre dans des dataframes. Vous risquez d'avoir besoin de fonctions spécifiques, uniquement disponible dans certains packages. N'hésitez pas à regarder sur Google pour vous aider !  
Les 3 fichiers à importer :
> prima.txt  
> mar2012.csv  

#### 2. Afficher les données

In [ ]:
iris

Sepal.Length  Sepal.Width  Petal.Length  Petal.Width     Species
1             5.1          3.5           1.4          0.2      setosa
2             4.9          3.0           1.4          0.2      setosa
3             4.7          3.2           1.3          0.2      setosa
4             4.6          3.1           1.5          0.2      setosa
5             5.0          3.6           1.4          0.2      setosa
6             5.4          3.9           1.7          0.4      setosa
7             4.6          3.4           1.4          0.3      setosa
8             5.0          3.4           1.5          0.2      setosa
9             4.4          2.9           1.4          0.2      setosa
10            4.9          3.1           1.5          0.1      setosa
11            5.4          3.7           1.5          0.2      setosa
12            4.8          3.4           1.6          0.2      setosa
13            4.8          3.0           1.4          0.1      setosa
14            4.3          3.0           1.1          0.1      setosa
15            5.8          4.0           1.2          0.2      setosa
16            5.7          4.4           1.5          0.4      setosa
17            5.4          3.9           1.3          0.4      setosa
18            5.1          3.5           1.4          0.3      setosa
19            5.7          3.8           1.7          0.3      setosa
20            5.1          3.8           1.5          0.3      setosa
21            5.4          3.4           1.7          0.2      setosa
22            5.1          3.7           1.5          0.4      setosa
23            4.6          3.6           1.0          0.2      setosa
24            5.1          3.3           1.7          0.5      setosa
25            4.8          3.4           1.9          0.2      setosa
26            5.0          3.0           1.6          0.2      setosa
27            5.0          3.4           1.6          0.4      setosa
28            5.2          3.5           1.5          0.2      setosa
29            5.2          3.4           1.4          0.2      setosa
30            4.7          3.2           1.6          0.2      setosa
31            4.8          3.1           1.6          0.2      setosa
32            5.4          3.4           1.5          0.4      setosa
33            5.2          4.1           1.5          0.1      setosa
34            5.5          4.2           1.4          0.2      setosa
35            4.9          3.1           1.5          0.2      setosa
36            5.0          3.2           1.2          0.2      setosa
37            5.5          3.5           1.3          0.2      setosa
38            4.9          3.6           1.4          0.1      setosa
39            4.4          3.0           1.3          0.2      setosa
40            5.1          3.4           1.5          0.2      setosa
41            5.0          3.5           1.3          0.3      setosa
42            4.5          2.3           1.3          0.3      setosa
43            4.4          3.2           1.3          0.2      setosa
44            5.0          3.5           1.6          0.6      setosa
45            5.1          3.8           1.9          0.4      setosa
46            4.8          3.0           1.4          0.3      setosa
47            5.1          3.8           1.6          0.2      setosa
48            4.6          3.2           1.4          0.2      setosa
49            5.3          3.7           1.5          0.2      setosa
50            5.0          3.3           1.4          0.2      setosa
51            7.0          3.2           4.7          1.4  versicolor
52            6.4          3.2           4.5          1.5  versicolor
53            6.9          3.1           4.9          1.5  versicolor
54            5.5          2.3           4.0          1.3  versicolor
55            6.5          2.8           4.6          1.5  versicolor
56            5.7          2.8           4.5          1.3  versicolor
57            6

Utiliser les fonctions head() et tail() pour afficher les 5 premières et 5 dernières lignes du dataset

❓ Quelle est la taille du dataset ? Utiliser la fonction shape et interpréter le résultat

Afficher toutes les colonnes du dataset avec un 'columns'

Sélectionner uniquement la colonnes Sepal.Length   
Vous pouvez vous inspirer de cette documentation : [Aide](https://apprendrepython.com/pandas-selectionner-des-lignes-colonnes-dans-un-dataframe-en-indexant/)

Regarder le nombre de fleurs par variétés que contient le dataset (utiliser la fonction [value_counts](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.value_counts.html) )

❓ Quelle est la largeur moyenne du sépal ?

❓ Quelle est la valeur maximale de longueur de pétale ?

Sélectionner uniquement les données pour lesquelles la longueur du pétal est >= 4.1 et stocker le résultat dans une nouvelle variable

#### 3. Export des données

**Pour écrire des données dans un fichier texte (`.txt`) :**

data = ["Ligne 1", "Ligne 2", "Ligne 3"]

with open("data.txt", "w") as file:  
        file.write("\n".join(data))
    

**Pour écrire des données dans un fichier csv (`.csv`) :**

import pandas as pd

df = pd.DataFrame(data[1:], columns=data[0])  
df.to_csv("data.csv", index=False)


**Pour écrire des données dans un fichier Excel (`.xlsx`) :**

import pandas as pd  
df = pd.DataFrame(data)  
df.to_excel("data.xlsx", index=False)